bpf/bpf_host.c

```diff
@@ -1127,4 +1132,120 @@ int to_host(struct __ctx_buff *ctx)
 	return ret;
 }
 
+#if defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_ROUTING)
+static __always_inline int
+/* Handles packet from a local endpoint entering the host namespace. Applies
+ * ingress host policies.
+ */
+to_host_from_lxc(struct __ctx_buff *ctx __maybe_unused)
+{
+	int ret = CTX_ACT_OK;
+	__u32 src_id = 0;
+	__u16 proto = 0;
+
+	if (!validate_ethertype(ctx, &proto)) {
+		ret = DROP_UNSUPPORTED_L2;
+		goto out;
+	}
+
+	switch (proto) {
+# if defined ENABLE_ARP_PASSTHROUGH || defined ENABLE_ARP_RESPONDER
```

Comment:
Is this case here relevant for ENABLE_ARP_RESPONDER?

Reply:
I copy pasted the equivalent code from the `to-host` program. Should it be different for traffic from lxc devices?

Reply:
Ok I think it's good. Originally we only ARP-responded to gw address for traffic from lxc, but it seems unknown IPs we just pass up the stack, hence we'd also have to let this pass through here as well.